rdt-app/src/main/java/io/ona/rdt_app/util/RDTJsonFormUtils.java

```diff
@@ -0,0 +1,156 @@
+package io.ona.rdt_app.util;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.AsyncTask;
+import android.util.Log;
+import android.widget.Toast;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.domain.ProfileImage;
+import org.smartregister.repository.ImageRepository;
+import org.smartregister.util.AssetHandler;
+import org.smartregister.view.activity.DrishtiApplication;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.UUID;
+
+import edu.washington.cs.ubicomplab.rdt_reader.ImageUtil;
+import edu.washington.cs.ubicomplab.rdt_reader.callback.OnImageSavedCallBack;
+import io.ona.rdt_app.activity.RDTJsonFormActivity;
+import io.ona.rdt_app.application.RDTApplication;
+
+import static io.ona.rdt_app.util.Constants.JSON_FORM_PARAM_JSON;
+import static io.ona.rdt_app.util.Constants.PROFILE_PIC;
+import static io.ona.rdt_app.util.Constants.REQUEST_CODE_GET_JSON;
+import static org.smartregister.util.JsonFormUtils.ENTITY_ID;
+
+/**
+ * Created by Vincent Karuri on 24/05/2019
+ */
+public class RDTJsonFormUtils {
+
+    private static final String TAG = RDTJsonFormUtils.class.getName();
+
+    private void startJsonForm(JSONObject form, Activity context, int requestCode) {
+        Intent intent = new Intent(context, RDTJsonFormActivity.class);
+        try {
+            intent.putExtra(JSON_FORM_PARAM_JSON, form.toString());
+            context.startActivityForResult(intent, requestCode);
+        } catch (Exception e) {
+            Log.e(TAG, e.getMessage());
+        }
+    }
+
+    public JSONObject getFormJsonObject(String formName, Context context) throws JSONException {
+        String formString = AssetHandler.readFileFromAssetsFolder(formName, context);
+        return new JSONObject(formString);
+    }
+
+    public void showToast(final Activity activity, final String text) {
+        if (activity != null) {
+            activity.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    Toast.makeText(activity, text, Toast.LENGTH_LONG).show();
+                }
+            });
+        }
+    }
+
+    public static void saveStaticImageToDisk(final Context context, final Bitmap image, final String providerId, final String entityId, final OnImageSavedCallBack onImageSavedCallBack) {
+        if (image == null || StringUtils.isBlank(providerId) || StringUtils.isBlank(entityId)) {
+            onImageSavedCallBack.onImageSaved(null);
```

Comment:
Right now, if you do not invoke the callback, the rdt capture activity launched is not finished (doesn't close out after capturing and saving the image).

I wanted to avoid having to change the rdt capture library to fix this, so this is a work-around.